tests/ecosystem/upgrade/test_noobaa.py

```diff
@@ -161,4 +162,27 @@ def test_noobaa_postupgrade(
         LOCAL_TEMP_PATH,
         mcg_obj_session
     )
-    assert bucket.status == constants.STATUS_BOUND
+    assert bucket.phase == constants.STATUS_BOUND
+
+
+@aws_platform_required
+@bugzilla('1820974')
+@pre_upgrade
+def test_buckets_before_upgrade(upgrade_buckets, mcg_obj_session):
+    """
+    Test that all buckets in cluster are in OPTIMAL state before upgrade.
+    """
+    for bucket in mcg_obj_session.read_system().get('buckets'):
+        assert bucket.get('mode') == BS_OPTIMAL
+
+
+@aws_platform_required
+@bugzilla('1820974')
+@post_upgrade
+@pytest.mark.polarion_id("OCS-2181")
+def test_buckets_after_upgrade(upgrade_buckets, mcg_obj_session):
+    """
+    Test that all buckets in cluster are in OPTIMAL state after upgrade.
+    """
+    for bucket in mcg_obj_session.read_system().get('buckets'):
+        assert bucket.get('mode') == BS_OPTIMAL
```

Comment:
Do you plant do add a check that the buckets are still actually there (here or in another test)?

Reply:
Good idea, I will add a check for that.

Reply:
But the check might be more complicated and would require another test case. I will add it in a different PR.

Reply:
Issue for this: https://github.com/red-hat-storage/ocs-ci/issues/1908

Reply:
Ack.